File: 2018/11/bela/bela.py
import sys
from functools import reduce

lines = sys.stdin.readlines()


def get_db():
    db = {}
    text = """
        A,11,11
        K,4,4
        Q,3,3
        J,20,2
        T,10,10
        9,14,0
        8,0,0
        7,0,0
    """
    lines = list(map(lambda s: s.strip(), text.split('\n')))[1:-1]
    for line in lines:
        c, a, b = line.split(',')
        db[c] = [int(a), int(b)]
    return db


db = get_db()
_, dom = lines[0].strip().split(' ')

output = sum(map(lambda line: db[line[0]][line[1] != dom], lines[1:]))
print(output)

File: 2018/11/bela/bela.js
/*eslint no-console: "off"*/

const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
  terminal: false
});

const lines = [];

rl.on("line", line => {
  lines.push(line);
});

rl.on("close", () => {
  const db = getDb();
  const [, dom] = lines[0].trim().split(" ");
  const output = lines
    .slice(1)
    .map(line => db[line[0]][Number(line[1] !== dom)])
    .reduce((a, b) => a + b);

  console.log(output);
});

function getDb() {
  const db = {};
  const text = `
      A,11,11
      K,4,4
      Q,3,3
      J,20,2
      T,10,10
      9,14,0
      8,0,0
      7,0,0
  `;
  const lines = text
    .split("\n")
    .map(s => s.trim())
    .slice(1, -1);

  for (const line of lines) {
    const [c, a, b] = line.split(",");
    db[c] = [parseInt(a), parseInt(b)];
  }

  return db;
}
